AutomatedTesting/Gem/PythonTests/AWS/Windows/core/test_aws_resource_interaction.py

```diff
@@ -57,130 +88,57 @@ def setup(launcher: pytest.fixture, cdk: pytest.fixture, resource_mappings: pyte
-                           resource_mappings: pytest.fixture,
-                           workspace: pytest.fixture,
-                           asset_processor: pytest.fixture
-                           ):
-        """
-        Setup: Deploys the CDK.
-        Tests: Runs the test level.
-        Verification: Searches the logs for the expected output from the example lambda.
-        """
-
-        log_monitor = setup(launcher, cdk, resource_mappings, asset_processor)
-
-        launcher.args = ['+LoadLevel', level]
-        launcher.args.extend(['-rhi=null'])
-
-        with launcher.start(launch_ap=False):
-            result = log_monitor.monitor_log_for_lines(
-                expected_lines=['(Script) - [Lambda] Completed Invoke',
-                                '(Script) - [Lambda] Invoke success: {"statusCode": 200, "body": {}}'],
-                unexpected_lines=['(Script) - Request validation failed, output file miss full path.',
-                                  '(Script) - '],
-                halt_on_unexpected=True
-            )
-
-        assert result
-
-    def test_get_dynamodb_value(self,
+    @pytest.mark.parametrize('expected_lines', [
+        ['(Script) - [S3] Head object request is done',
+         '(Script) - [S3] Head object success: Object example.txt is found.',
+         '(Script) - [S3] Get object success: Object example.txt is downloaded.',
+         '(Script) - [Lambda] Completed Invoke',
+         '(Script) - [Lambda] Invoke success: {"statusCode": 200, "body": {}}',
+         '(Script) - [DynamoDB] Results finished']])
+    @pytest.mark.parametrize('unexpected_lines', [
+        ['(Script) - [S3] Head object error: No response body.',
+         '(Script) - [S3] Get object error: Request validation failed, output file directory doesn\'t exist.',
+         '(Script) - Request validation failed, output file miss full path.',
+         '(Script) - ']])
+    def test_scripting_behavior(self,
                                 level: str,
                                 launcher: pytest.fixture,
-                                cdk: pytest.fixture,
-                                resource_mappings: pytest.fixture,
                                 workspace: pytest.fixture,
                                 asset_processor: pytest.fixture,
+                                resource_mappings: pytest.fixture,
                                 aws_utils: pytest.fixture,
-                                ):
+                                expected_lines: typing.List[str],
+                                unexpected_lines: typing.List[str]):
         """
-        Setup: Deploys  the CDK application
-        Test: Runs a launcher with a level that loads a scriptcanvas that pulls a DynamoDB table value.
-        Verification: The value is output in the logs and verified by the test.
+        Setup: Deploys cdk and updates resource mapping file.
+        Tests: Getting AWS credentials for no signed in user.
```

Comment:
nit: "no signed in user"? Maybe "Getting credentials when no credentials are configured"